x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -35,7 +35,7 @@
     private final int totalShards;
     private final int skippedShards;
     private final Clusters clusters;
-    private final AtomicArray<ShardSearchFailure> shardFailures;
+    private final AtomicArray<ShardSearchFailure> queryFailures;
```

Comment:
Can we keep the name as is for now. I understand that we want to call it `query` failures after this change but we may also want to provide a way to return partial top hits in the future. My point being that we might reintroduce fetch failures when they make sense.

Reply:
this is not a huge deal, but can we not rename it back to shardFailures in the future once we introduce a way to return partial top hits?

Reply:
As you want, that's not a big deal I agree but I wanted to raise the awareness that returning partial top hits is still in the air.

Reply:
Thanks, I am aware, and I probably would have made the same comment if you had made this same change :)